src/main/fc/fc_core.c

```diff
@@ -295,6 +295,14 @@ static void updateArmingStatus(void)
 	       DISABLE_ARMING_FLAG(ARMING_DISABLED_SERVO_AUTOTRIM);
 	    }
 
+        if (isModeActivationConditionPresent(BOXPREARM)) {
+            if (IS_RC_MODE_ACTIVE(BOXPREARM)) {
+                DISABLE_ARMING_FLAG(ARMING_DISABLED_NO_PREARM);
```

Comment:
We should also disable `ARMING_DISABLED_NO_PREARM` if `BOXPREARM` is not configured. This way we won't end up in a situation where a person configuring the aircraft will get stuck being unable to arm if they configure and then remove the `PREARM` mode.

Reply:
Good catch! Didn't think of that.